ocaml/typing/typecore.ml

```diff
@@ -3679,7 +3684,13 @@ let type_pattern_approx env spat ty_expected =
       end;
   | _ -> ()
 
-let rec type_function_approx env loc label spato sexp in_function ty_expected =
+let wrap_rhs = function
+  | Psimple_rhs e -> `Expr e
+  | Pboolean_guarded_rhs { pbg_rhs; _ } -> `Expr pbg_rhs
+  | Ppattern_guarded_rhs { ppg_scrutinee; ppg_cases; ppg_loc } ->
+      `Desc (Pexp_match (ppg_scrutinee, ppg_cases), ppg_loc)
```

Comment:
This helper function is gross, and we'd need to document it if we kept it around. Can we get rid of this polymorphic variant type and instead just pass the whole rhs around?

Reply:
Agreed, I removed it, which required us to instead wrap an `expression` in a `Psimple_rhs`, but this is more intuitive (the expression is in an rhs-like role) and simpler.